### ISSUE TITLE
gsl wants libgsl >= 1.16 but I have 2.4

### ISSUE THREAD

**Issue by nzok** [RobinHankin/gsl#18]
Script started on 2021-11-23 15:26:59+1300
% R

R version 3.4.4 (2018-03-15) -- "Someone to Lean On"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("gsl")
Installing package into ‘/home/ok/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/gsl_2.1-7.1.tar.gz'
Content type 'application/x-gzip' length 222357 bytes (217 KB)
==================================================
downloaded 217 KB

* installing *source* package ‘gsl’ ...
** package ‘gsl’ successfully unpacked and MD5 sums checked
checking for gsl-config... /usr/bin/gsl-config
checking if GSL version >= 2.1... checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
configure: error: Need GSL version >= 1.16
ERROR: configuration failed for package ‘gsl’
* removing ‘/home/ok/R/x86_64-pc-linux-gnu-library/3.4/gsl’

The downloaded source packages are in
    ‘/tmp/RtmpRIIADX/downloaded_packages’
Warning message:
In install.packages("gsl") :
  installation of package ‘gsl’ had non-zero exit status
> quit()
Save workspace image? [y/n/c]: n
$ gsl-config --version
2.4
$ uname -a
Linux tarski 5.4.0-90-generic #101~18.04.1-Ubuntu SMP Fri Oct 22 09:25:04 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
exit
Script done on 2021-11-23 15:27:42+1300

This was the most recent version of libgsl-dev available for Ubuntu 18.04

**Comment by markgalassi** [comment]
> This was the most recent version of libgsl-dev available for Ubuntu 18.04

Dear GSL users in R,

I'm delighted to hear of your using gsl, and I happened across your report.

I took a peak at the tarball you pointed to and found that it had some surprising code to find gsl in its configure.ac - I'm not surprised that it did not return something useful to configure.

I quickly put together an autoconf+automake setup to find gsl in what might be the simplest way, using pkg-config.  Here's the snippet in configure.ac which I hope can help you:

AC_PATH_TOOL(PKG_CONFIG, pkg-config)
AC_SUBST([GSL_CFLAGS])
AC_SUBST([GSL_LIBS])
PKG_CHECK_MODULES([GSL], [gsl >= 2.4])
CFLAGS="$GSL_CFLAGS $CFLAGS"
LIBS="$GSL_LIBS $LIBS"

On an ubuntu-18.04 system with libgsl-dev installed it gives this:

checking pkg-config is at least version 0.9.0... yes
checking for gsl... yes

and it links with the right libraries (pkg-config gsl --cflags does not need or set C flags).

And if I remove libgsl-dev then it reports the stuff below:

checking for pkg-config... /usr/bin/pkg-config
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for gsl... no
configure: error: Package requirements (gsl >= 2.4) were not met:

No package 'gsl' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables gsl_CFLAGS
and gsl_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

**Comment by sherbert** [comment]
Just ran into this and spent some time debugging. The requirement actually being checked for in `configure` is `libgsl-dev >= 2.5`:
```
cat confdefs.h - <<_ACEOF >conftest.$ac_ext
/* end confdefs.h.  */

#include <string.h>
#include <gsl/gsl_version.h>
#include <stdio.h>
#include <stdlib.h>
int main() {
#ifdef GSL_VERSION
   int major, minor;
   char *gslv = GSL_VERSION;
   if ((sscanf(gslv, "%d.%d", &major, &minor)) != 2) {
     exit (1);
   }
   exit (minor < 5);
   exit (major < 2);
#else
  exit(1);
#endif
}
```
The error message when this check fails needs to be updated to reference the version actually being checked for rather than `1.16`:
```
  as_fn_error $? "Need GSL version >= 1.16" "$LINENO" 5
```

**Comment by Japrin** [comment]
The same problem. I have to use an older version of gsl ([gsl_2.1-6.tar.gz](https://cran.r-project.org/src/contrib/Archive/gsl/gsl_2.1-6.tar.gz)

**Comment by mmaechler** [comment]
This was partly addressed in #17   but indeed, not fully.    [There](https://github.com/RobinHankin/gsl/issues/17#issuecomment-1132110642), you have a hint on *how* to test correctly for 2.5 {but the hint is not directly usable: you need `exit(.)` codes, not return}.


BTW: It is `configure.ac`  which must be changed manually, then (typically you rebuild `configure` from `configure.ac`
